js/style/style_declaration.js

```diff
@@ -23,20 +23,31 @@ function StyleDeclaration(propType, renderType, prop, value) {
 
 }
 
-StyleDeclaration.prototype.calculate = function(z) {
-    return typeof this.value === 'function' ? this.value(z) : this.value;
+StyleDeclaration.prototype.calculate = function(z, zoomHistory) {
+    return typeof this.value === 'function' ? this.value(z, zoomHistory) : this.value;
 };
 
-StyleDeclaration.prototype.parseValue = function(value, type, values) {
+StyleDeclaration.prototype.parseValue = function(value, type, values, prop) {
     return type === 'color' ? parseColor(value) :
         type === 'number' ? parseNumber(value) :
         type === 'boolean' ? Boolean(value) :
         type === 'image' ? String(value) :
-        type === 'string' ? String(value) :
+        type === 'string' ? parsePattern(value) :
+        type === 'array' && prop === 'line-dasharray' ? parsePattern(value) :
```

Comment:
@jfirebaugh I added functions for all the other string properties, but I think we'll need to make dasharrays a different type than other number arrays. Other number arrays would use interpolated stops functions instead of faded stops functions.